pkg/resource/resource_suppliers.go

```diff
@@ -1,11 +1,19 @@
 package resource
 
-var resourceSupplier = make([]Supplier, 0)
+type SupplierLibrary struct {
+	resourceSupplier []Supplier
+}
+
+func NewLibrary() *SupplierLibrary {
```

Comment:
I've hesitated, I'm not a big fan on my own naming, if you have better idea ...

I'm ok to rename theses methods to avoid collisions if we stay on "Library" naming

Reply:
I don't have any preferences on the naming

Reply:
Would be better to have supplier in the name cause we will call `resource.NewLibrary` which is missleading for me (it's not a library of resources)